cli/command/image/build.go

```diff
@@ -361,7 +361,7 @@ func runBuild(ctx context.Context, dockerCli command.Cli, options buildOptions)
 		}
 	}
 
-	err = jsonmessage.DisplayJSONMessagesStream(response.Body, buildBuff, dockerCli.Out().FD(), dockerCli.Out().IsTerminal(), aux)
+	err = jsonmessage.DisplayStreamWithTerminal(ctx, response.Body, buildBuff, dockerCli.Out().FD(), dockerCli.Out().IsTerminal(), aux)
```

Comment:
Can't we just use `DisplayStream` here?

```suggestion
	err = jsonmessage.DisplayStream(ctx, response.Body, buildBuff, dockerCli.Out(), aux)
```

This way we don't need `DisplayStreamWithTerminal` at all.

Reply:
I'm not sure, I guess we could? I wasn't sure if this would break something, but can give it a try

Reply:
yeah, I was checking the code a bit and we would need to change this line to a `*streams.Out` then wrap it back to `io.Writer` on `fmt.Sprintf` https://github.com/docker/cli/pull/5663/files/bfd3131c33c56a973b7f9976e929633fa0cf548b#diff-543eeee334b558064e83e500fd02a15a21f142759fbd573adec500a24b088c6bR194

Reply:
`*streams.Out` satisfies the `io.Writer` interface, so I don't think we need to cast it back.

So this should work I guess?
```suggestion
	err = jsonmessage.DisplayStream(ctx, response.Body, streams.NewOut(buildBuff), dockerCli.Out(), aux)
```

Reply:
This will work

Reply:
okay, removed the `DisplayWithTerminal` function and applied your suggestion